rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/filter/UserLoginFilter.java

```diff
@@ -166,28 +173,72 @@ private String getKualiSessionId(final Cookie[] cookies) {
     /**
      * establishes the backdoor user on the established user id if backdoor capabilities are valid.
      */
-    private void establishBackdoorUser(HttpServletRequest request) {
+    protected void establishBackdoorUser(HttpServletRequest request) {
         final String backdoor = request.getParameter(KRADConstants.BACKDOOR_PARAMETER);
-        if (StringUtils.isNotBlank(backdoor)) {
-            if (!getKualiConfigurationService().getPropertyValueAsString(KRADConstants.PROD_ENVIRONMENT_CODE_KEY)
-                    .equalsIgnoreCase(getKualiConfigurationService().getPropertyValueAsString(
-                            KRADConstants.ENVIRONMENT_KEY))) {
-                if (getParameterService().getParameterValueAsBoolean(KRADConstants.KUALI_RICE_WORKFLOW_NAMESPACE,
-                        KRADConstants.DetailTypes.BACKDOOR_DETAIL_TYPE, KewApiConstants.SHOW_BACK_DOOR_LOGIN_IND)) {
-                    try {
-                        KRADUtils.getUserSessionFromRequest(request).setBackdoorUser(backdoor);
-                    } catch (RiceRuntimeException re) {
-                        //Ignore so BackdoorAction can redirect to invalid_backdoor_portal
-                    }
-                }
-            }
+        UserSession userSession = KRADUtils.getUserSessionFromRequest(request);
+
+        if (StringUtils.isNotBlank(backdoor) && !isProductionEnvironment() && showBackdoorLogin() &&
+                isBackdoorAuthorized(userSession)) {
+
+            KRADUtils.getUserSessionFromRequest(request).setBackdoorUser(backdoor);

```

Comment:
I am not seeing why it would needed to be wrapped with a RiceRuntimeException

  public static final UserSession getUserSessionFromRequest(HttpServletRequest request) {
        return (UserSession) request.getSession().getAttribute(KRADConstants.USER_SESSION_KEY);
    }